pkg/archive/untar.go

```diff
@@ -24,6 +24,12 @@ func Untar(r io.Reader, dir string, gzip bool, quiet bool) error {
 	return untar(r, dir, gzip, quiet)
 }
 
+// UntarXZ reads the (xz-compressed) tar file from r and writes it into dir.
+// This is a wrapper for the implementation in xz.go
+func UntarXZ(r io.Reader, dir string, quiet bool) error {
+	return untarXZ(r, dir, quiet)
+}
+
```

Comment:
Can you push that function to archive/xz.go itself so that all the xz decompression can be part of single file like this file is more like tar.gz

Reply:
I think the better way of doing it just extend the `untar` function.

```golang
func untar(r io.Reader, dir string, gzip bool, xz bool, quiet bool) (err error) {
	[...]
	if gzip {
		r, err = gzipc.NewReader(r)
		if err != nil {
			return fmt.Errorf("requires gzip-compressed body: %v", err)
		}
         }
         if xz {
               r, err := xz.NewReader(r)
	       if err != nil {
		  return fmt.Errorf("requires xz-compressed body: %v", err)
	       }
         }
 [...]
```
if we really want to go bit more refactor then I think untar should just detect the file type and according to that 
something like 
```golang
switch {
	case filetype.Is(header, "xz"):
....
```